homeassistant/components/switchbot_cloud/lock.py

```diff
@@ -37,6 +38,13 @@ def _handle_coordinator_update(self) -> None:
         """Handle updated data from the coordinator."""
         if coord_data := self.coordinator.data:
             self._attr_is_locked = coord_data["lockState"] == "locked"
+            self._attr_is_locking = coord_data["lockState"] == "locking"
```

Comment:
I guess the 3 new states only work for the Lock Pro but do not impact the classic version.

I do not know how the extra state attribute work.

As the coordinator refreshes [only once every 10 minutes because it is polling data](https://www.home-assistant.io/integrations/switchbot_cloud/), I am not sure if the locking and unlocking state will be useful. I saw a PR trying to implement webhooks to get real time updates though.

Did you test locally ? I feel that something might be missing to support the Lock Pro but I might be mistaken. The documentation also needs to be updated to say that the Lock Pro is supported.

Reply:
@SeraphicRav if you have a link to the PR that avoids polling, I'm happy to take a look as well. Thanks for your review!

Reply:
My mistake, there is no PR, just an issue: https://github.com/home-assistant/core/issues/129317

Reply:
Ok, no issue with the "Lock" version:
![image](https://github.com/user-attachments/assets/fb80186a-3f71-4d5d-86ae-1623d133ebe3)
Seems good to me ! hanks ;)

Reply:
The door state is not updating in real time as expected. Maybe a separate entity for it would be better ? Cannot use it as a sensor.

Reply:
Same comment for the battery.

Reply:
So a separate BinarySensor for door state and SensorDeviceClass.BATTERY for the battery? Make sense, I just have to figure out how to integrate that into the lock device properly.

Reply:
I see. Let us see what an Home Assistant owner will say. :)